### ISSUE TITLE
[SPIR-V] Passing a buffer array to a function causes crash

### ISSUE THREAD

**Issue by azhirnov** [microsoft/DirectXShaderCompiler#3790]
This code compiles successfully for DXIL, but cause crash when compiling for SPIRV.
```cpp
float4 ReadBuffer(StructuredBuffer<float4> Buffers[2], uint idx)
{
    return Buffers[idx][1];
}

StructuredBuffer<float4> g_Buffers[2];
RWTexture2D<float4>      g_OutImage;

[numthreads(8, 8, 1)]
void CSMain(uint2 DTid : SV_DispatchThreadID)
{
    g_OutImage[DTid] = ReadBuffer(g_Buffers, DTid.x);
}
```
Console command:
```
dxc -spirv -T cs_6_5 -E CSMain test.hlsl
Internal compiler error: access violation. Attempted to read from address 0x000000000000002C
```

**Comment by jaebaek** [comment]
@azhirnov Thank you for reporting this issue. We will take a look!

**Comment by sudonatalie** [comment]
I believe the crash mentioned in this issue was resolved by #4188, but the provided sample shader fails to compile for both DXIL and SPIR-V with the same (seemingly legitimate) error: 

```
test.hlsl:1:44: error: implicit instantiation of template 'StructuredBuffer<vector<float, 4> >' within its own definition
float4 ReadBuffer(StructuredBuffer<float4> Buffers[2], uint idx)
                                           ^
```

Closing as I think the crash has been resolved, but if this is still an issue, please re-open with an updated repro shader.
